File: 2149-remove-colored-pieces-if-both-neighbors-are-the-same-color/2149-remove-colored-pieces-if-both-neighbors-are-the-same-color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        cntA = 0
        cntB = 0
        for i in range(1, len(colors) - 1):
            if colors[i - 1] == colors[i] == colors[i + 1]:
                if colors[i] == 'A':
                    cntA += 1
                if colors[i] == 'B':
                    cntB += 1
        return cntA > cntB

File: 2149-remove-colored-pieces-if-both-neighbors-are-the-same-color/2149-remove-colored-pieces-if-both-neighbors-are-the-same-color.java
class Solution {
    public boolean winnerOfGame(String colors) {
        char[] c = colors.toCharArray();
        int n = c.length, cntA = 0, cntB = 0;
        for(int i = 1; i < n - 1; i++) {
            if(c[i] == 'A' && c[i - 1] == 'A' && c[i + 1] == 'A') cntA++;
            if(c[i] == 'B' && c[i - 1] == 'B' && c[i + 1] == 'B') cntB++;
        }
        return cntA > cntB;
    }
}